backend/app/views/spree/admin/promotions/_form.html.erb

```diff
@@ -8,17 +8,10 @@
           <%= f.text_field :name, :class => 'fullwidth' %>
         <% end %>
 
-        <%= fields_for :promotion_builder do |bf| %>
-          <%= bf.field_container :base_code do %>
-            <%= bf.label :base_code %>
-            <%= bf.text_field :base_code, :readonly => !@promotion.new_record?, :class => 'fullwidth' %>
-          <% end %>
-
-          <%= bf.field_container :number_of_codes do %>
-            <%= bf.label :number_of_codes %>
-            <%= bf.text_field :number_of_codes, :readonly => !@promotion.new_record?, class: 'fullwidth' %>
-          <% end %>
-        <% end %>
+        <div class="field" id="promotion_single_code_field">
```

Comment:
Since we don't use this ID anywhere we should rewrite this into a `data-hook` attribute, if I assume correctly that you added this for all the defacers out there.